tmp directory not created

With the latest addition of TMPDIR having `/tmp` appended, the directory doesn't exist and the build fails.

```
E               subprocess.CalledProcessError: Command 'PATH=/home/rookcheck/tmp/rookcheck/rookcheck-rookcheck-3478/rook_build/go/bin:$PATH GOPATH=/home/rookcheck/tmp/rookcheck/rookcheck-rookcheck-3478/rook_build TMPDIR=/home/rookcheck/tmp/rookcheck/tmp make --directory='/home/rookcheck/tmp/rookcheck/rookcheck-rookcheck-3478/rook_build/src/github.com/rook/rook' -j BUILD_REGISTRY='rook-build' IMAGES='ceph' build' returned non-zero exit status 2.

tests/lib/common.py:173: CalledProcessError
```

In my case, `/home/rookcheck/tmp/rookcheck` does get created, but `/home/rookcheck/tmp/rookcheck/tmp` does not.  If I create the directory manually and rerun `tox -e py37 tests/test_basic.py`, then the test completes successfully.

--- a/tests/lib/rook/upstream.py
+++ b/tests/lib/rook/upstream.py
@@ -18,7 +18,6 @@ import re
 import time
 import wget
 
-from tests.config import settings
 from tests.lib.common import execute
 from tests.lib import common
 from tests.lib.rook.base import RookBase
@@ -32,6 +31,8 @@ class RookCluster(RookBase):
         self._rook_built = False
         self.builddir = os.path.join(self.workspace.working_dir, 'rook_build')
         os.mkdir(self.builddir)
+        self.go_tmpdir = os.path.join(self.workspace.working_dir, 'tmp')
+        os.mkdir(self.go_tmpdir)
 
     def build_rook(self):
         logger.info("[build_rook] Download go")
@@ -72,7 +73,7 @@ class RookCluster(RookBase):
             "make --directory='{builddir}/src/github.com/rook/rook' "
             "-j BUILD_REGISTRY='rook-build' IMAGES='ceph' "
             "build".format(builddir=self.builddir,
-                           tmpdir=os.path.join(settings.WORKSPACE_DIR, 'tmp')),
+                           tmpdir=self.go_tmpdir),
             log_stderr=False,
             logger_name="make -j BUILD_REGISTRY='rook-build' IMAGES='ceph'",
         )

